### ISSUE TITLE
(e621) Authentification is required despite having Authentication set up

### ISSUE THREAD

**Issue by lyssieth** [Xwilarg/BooruSharp#3]
I have authentication set up for E621, but it keeps throwing this error.

**Comment by Xwilarg** [comment]
Hi,

E621 changed their API a lot recently, I'm currently working on a new version of BooruSharp to fix all of that, I'll tell you once it's done.

**Comment by lyssieth** [comment]
Ah, alright. Thank you!

**Comment by Xwilarg** [comment]
I finally updated BooruSharp, you can find the version 2.0.0 on NuGet: https://www.nuget.org/packages/BooruSharp/2.0.0

I changed a lot of things, you can have more information about that here: https://boorusharp.zirk.eu/transition.html
And you can find the documentation for the new version here: https://boorusharp.zirk.eu/
(I didn't finish writing it yet)

If you have any question don't hesitate to ask

**Comment by lyssieth** [comment]
Getting a `There is nothing available with these tags` error when trying to use `GetRandomImageAsync()` with E621. I passed various tags in separate calls (`dragon`, nothing at all, `wolf`, etc)

**Comment by Xwilarg** [comment]
Would you mind showing me your code?

I tried to reproduce the issue by doing the following:
```cs
static void Main(string[] args)
{
    var booru = new E621();
    GetImage(booru, "dragon").GetAwaiter().GetResult();
    GetImage(booru, "").GetAwaiter().GetResult();
    GetImage(booru, "wolf").GetAwaiter().GetResult();
    Console.Read();
}

private static async Task GetImage(ABooru booru, string tag)
{
        Console.WriteLine((await booru.GetRandomImageAsync(tag)).fileUrl);
}
```
But I don't get any error.

**Comment by lyssieth** [comment]
`GetAuth()` returns a BooruAuth with the auth details. I can gist my full class too if needed.
`Log` is my own logging API that makes a pretty Console.WriteLine
I return `SearchResult` instead of the URL because I want to have all the data available, so I can use it for crediting/linking properly

This is my code for when I don't have any tags in the input
```cs
public async Task<SearchResult> Random() {
	var client = new E621(GetAuth());

	var post = await client.GetRandomImageAsync();
	return post;
}
```

With tags
```cs
public async Task<SearchResult> Random(params string[] fullTags) {
    var tags = fullTags.Select(x => x.ToLower());
    var client = new E621(GetAuth());
    
    Log.Debug("Tags: " + string.Join('|', tags));
    var posts = await client.GetRandomImageAsync(fullTags.Select(x => x.ToLower()).ToArray());

    return posts;
}
```

**Comment by lyssieth** [comment]
Oh, and here's what calls it:
```cs
if (!string.IsNullOrWhiteSpace(tags))
	res = await UnsafeService.ESix.Random(tags);
else
	res = await UnsafeService.ESix.Random();
```
where `tags` is a space-separated tag list, or empty/whitespace

**Comment by Xwilarg** [comment]
There is indeed a problem with authentification, it is however not needed to get posts so you can just do `new E621()`.

**Comment by lyssieth** [comment]
Works without authentication.

**Comment by Xwilarg** [comment]
Great,

As for the authentification for E621, the 2nd argument of BooruAuth will be your API token (you can get it here: https://e621.net/users/[yourId]/api_key/view )
Things will be fixed in next version of BooruSharp

**Comment by lyssieth** [comment]
Yeah that's exactly what I already have happening. It grabs them from a config file since the code is public.

**Comment by Xwilarg** [comment]
Authentification is fixed with [version 2.0.1](https://www.nuget.org/packages/BooruSharp/2.0.1).
For E621 you need to instantiate a new BooruAuth using BooruAuth.CreateFromApiKey(string login, string apiKey)
More information here: https://boorusharp.zirk.eu/#authentification

Since this problem is solved I'm closing the issue, feel free to open it again if you have any question
